src/main/java/mate/academy/bookdao/BookDaoImpl.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection
+                                .prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                int price = resultSet.getInt("price");
+                Book book = Book.of(id, title, price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                            Statement statement = connection.createStatement();
+                            ResultSet resultSet = statement.executeQuery(sql)) {
+            while (resultSet.next()) {
+                long id = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                int price = resultSet.getInt("price");
+                Book book = Book.of(id, title, price);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection
+                                .prepareStatement(sql)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setInt(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book not found");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book.getTitle(), e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            int affectedRow = statement.executeUpdate();
+            if (affectedRow < 1) {
```

Comment:
Don't use `affectedRow < 1` to check if the book was not found. Use `affectedRow == 0` to check if no rows were affected, which means the book was not found. This is more explicit and understandable.